wgsl/index.bs

```diff
@@ -3261,6 +3270,61 @@ See [[#module-constants]] and [[#function-scope-variables]].
   </xmp>
 </div>
 
+### `override` Declarations ### {#override-decls}
+
+An <dfn noexport>override declaration</dfn> specifies a name for a
+[=pipeline-overridable=] constant value.
+The value of a <dfn noexport>pipeline-overridable</dfn> constant is fixed at
+pipeline-creation time.
+The value is the one specified by the WebGPU pipeline-creation method, if
+specified, and otherwise is the value of its initializer expression.
+When an [=identifier=] use [=resolves=] to a override-declaration, the identifier denotes that value.
+`override`-declarations must meet the following restrictions:
+
+  * The declaration must only occur at [=module scope=].
+  * The declaration must have at least one of a declared type, an initializer
+      expression, or both.
+  * The declared type, if present, must be a [=scalar=].
+  * The initializer expression, if present, must:
+      * evaluate to a [=scalar=] type.
+      * evaluate to the declared type if it is present.
+      * be composed only [=syntax/const_expression|const_expressions=] or
+          expressions where all identifiers [=resolve=] to overridable constants.
```

Comment:
anything to be said about recursive dependencies of them constants?

Reply:
We already have a rule that there should be no cycles among declarations that was introduced as part of out of order module declarations.